### PR TITLE
[W4.4b][F11-A1]Wu Hao

### PR DESCRIPTION
**Pull request by Wuhao-ooo**
Address class now consists 4 classes to store personal data.
Some code adapted from #27 (I follow the same technique for using validation regex).

### CODE REVIEW

**Comment by Wuhao-ooo** [comment]
Work in progress.
Test input and expected output to be updated.

**Comment by Wuhao-ooo** [comment]
Finally pass all checks.
Ready for review.

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -8,13 +8,18 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in the format \"a/BLOCK, STREET, UNIT, POSTAL_CODE\" ";
```
Comment (karrui) [inline src/seedu/addressbook/data/person/Address.java:12]: Avoid lines longer than 110 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.

File: test/java/seedu/addressbook/commands/DeleteCommandTest.java [hunk 2]
```diff
@@ -33,13 +33,13 @@
     @Before
     public void setUp() throws Exception {
         Person johnDoe = new Person(new Name("John Doe"), new Phone("61234567", false),
-                new Email("john@doe.com", false), new Address("395C Ben Road", false), new UniqueTagList());
+                new Email("john@doe.com", false), new Address("111, 395C Ben Road, #111, 111111", false), new UniqueTagList());
```
Comment (karrui) [inline test/java/seedu/addressbook/commands/DeleteCommandTest.java:36]: Do note that if you are separating the parameters into multiple lines, the parameters should be aligned with each other. Sometimes the formatting style in IntelliJ is slightly different from what we expect.
e.g. 
```
Person johnDoe = new Person(new Name("John Doe"), 
                 new Phone("61234567", false),
                 ...
```

Please do make sure the line width is still not more than 110 chars

File: test/java/seedu/addressbook/data/AddressBookTest.java [hunk 3]
```diff
@@ -47,25 +47,25 @@ public void setUp() throws Exception {
...
 
         bobChaplin     = new Person(new Name("Bob Chaplin"),
                                     new Phone("94321500", false),
                                     new Email("bob@nusgreyhats.org", false),
-                                    new Address("9 Computing Drive", false),
+                                    new Address("222, 9 Computing Drive, #222, 222222", false),
                                     new UniqueTagList(tagMathematician));
 
         charlieDouglas = new Person(new Name("Charlie Douglas"),
                                     new Phone("98751365", false),
```
Comment (karrui) [inline test/java/seedu/addressbook/data/AddressBookTest.java:60]: As noted above, just like this format if the parameters are to be on new lines.

**Review by karrui — COMMENTED** [review]
Good attempt for this LO! Some comments added. Please close the PR after reading / implementing suggestions.

**Comment by karrui** [comment]
Also, please try to make commit headers less than 50 chars in character to avoid being truncated!

**Pull request closed without merging**